File: fb/209.minimum-size-subarray-sum.cpp
// Time O(n)
// Space O(1)
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int n = nums.size();
        int ans = INT_MAX;
        int start = 0;
        int sum = 0;
        for (int end = 0; end < n; end++) {
            sum += nums[end];
            while (sum >= s) {
                ans = min(ans, end + 1 - start);
                sum -= nums[start++];
            }
        }
        return (ans != INT_MAX) ? ans : 0;
    }
};

File: fb/209.minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        INI_MAX = float("infinity")
        localSum = 0
        minLen = INI_MAX
        left = 0
        
        for i in range(len(nums)):
            localSum += nums[i]
            while localSum >= target:
                minLen = min(minLen, i + 1 - left)
                localSum -= nums[left]
                left += 1
                
        return minLen if minLen != INI_MAX else 0
        
